docs/templates/function.js

```diff
@@ -0,0 +1,143 @@
+import hljs from "highlight.js";
+import { promises } from "fs";
+
+import { outPath } from "../utils/io.js";
+import { getPath } from "../utils/url.js";
+import { scrollTo } from "../utils/scripts.js";
+import pageTemplate from "./page.js";
+import last from "../../array/last.js";
+
+const { writeFile: writeFileAsync, mkdir: mkdirAsync } = promises;
+
+const breadcrumb = (name, pathParts) => {
+  const parts = ["", ...pathParts, name];
+
+  let resultParts = [];
+
+  for (let i = 1; i < parts.length; i++) {
+    const part = parts.slice(0, i);
+    resultParts.push(
+      `<a href="${getPath(part)}">${last(part) === "" ? "std" : last(part)}</a>`
+    );
+  }
+
+  return `<div class="breadcrumbs">${resultParts.join("<span>/</span>")}</div>`;
+};
+
+const nameFragment = (name, pathParts) => {
+  return name
+    ? `<h3 class="name"><a href="${getPath(pathParts, name)}">${name}</a></h3>`
+    : "";
+};
+
+const descriptionFragment = description => {
+  if (description && !description.startsWith("TODO")) {
+    return `<div class="description">
+        <span>${description}</span>
+      </div>`;
+  }
+
+  return "";
+};
+
+const signatureFragment = signature => {
+  if (signature && !signature.startsWith("TODO")) {
+    const singatureFormatted = hljs.highlight("typescript", signature).value;
+
+    return `<div class="type-signature">
+      <h3>Type signature</h3>
+      <div class="content">
+        <pre><code class="hljs"><span>${singatureFormatted}</span></code></pre>
+      </div>
+    </div>`;
+  }
+
+  return "";
+};
+
+const examplesFragment = (examples, pathParts, funcName) => {
+  if (
```

Comment:
Consider using early returns in every fragment - the nesting will be reduced substantially.